### PR TITLE
soc: nrf52: validate_enabled_instances fixes

### PR DESCRIPTION
**Pull request by mbolivar-nordic**
UART0, TWI0, and SPI0 have different peripheral IDs (2, 3, and 4
respectively) on this SoC and therefore should not be mutually
exclusive to use.

### CODE REVIEW

**Review by anangl — APPROVED** [review]

**Comment by mbolivar-nordic** [comment]
@ioannisg @carlescufi CI failures are unrelated; please merge

**Review by jfischer-no — APPROVED** [review]

**Review by ioannisg — COMMENTED** [review]
how about 52811, @mbolivar-nordic ?

**Comment by mbolivar-nordic** [comment]
> how about 52811, @mbolivar-nordic ?

I got a report from an FAE only about this SoC. I'm not trying to audit all of the SoCs and fix the file for them, just trying to address this particular bug report.

Are you saying you've checked everything and 52811 is the only other one missing?

**Comment by ioannisg** [comment]
> > how about 52811, @mbolivar-nordic ?
> 
> I got a report from an FAE only about this SoC. I'm not trying to audit all of the SoCs and fix the file for them, just trying to address this particular bug report.
> 
> Are you saying you've checked everything and 52811 is the only other one missing?

Did't check, but, afaik the three "low-end" nrf52x variants we have are 52805, 52810, 52811, so it is worth checking one more SoC.

**Review by carlescufi — APPROVED** [review]

**Comment by carlescufi** [comment]
@anangl could you please check that?

**Comment by mbolivar-nordic** [comment]
@ioannisg I added a fix for 52811 also.

**Review by ioannisg — APPROVED** [review]

File: soc/arm/nordic_nrf/validate_enabled_instances.c [hunk 1]
```diff
@@ -40,7 +42,9 @@
 	IF_ENABLED(CONFIG_SOC_SERIES_NRF91X, (", UARTE"#idx)) \
 	". Check nodes with status \"okay\" in zephyr.dts."
 
-#if !IS_ENABLED(CONFIG_SOC_NRF52810) && !IS_ENABLED(CONFIG_SOC_NRF52805)
+#if (!IS_ENABLED(CONFIG_SOC_NRF52810) &&	\
+	!IS_ENABLED(CONFIG_SOC_NRF52805) &&	\
+	!IS_ENABLED(CONFIG_SOC_NRF52811))
```
Comment (anangl) [inline soc/arm/nordic_nrf/validate_enabled_instances.c:47]: For nRF52811, we need to also add the following:
```
#if IS_ENABLED(CONFIG_SOC_NRF52811)
BUILD_ASSERT(!(SPI_ENABLED(1) && I2C_ENABLED(0)),
	     "Only one of the following peripherals can be enabled: "
	     "SPI1, SPIM1, SPIS1, TWI0, TWIM0, TWIS0. "
	     "Check nodes with status \"okay\" in zephyr.dts.");
#endif
```

**Review by anangl — CHANGES_REQUESTED** [review]

File: soc/arm/nordic_nrf/validate_enabled_instances.c (hunk 1 above)
Comment (mbolivar-nordic) [reply soc/arm/nordic_nrf/validate_enabled_instances.c:47]: > For nRF52811, we need to also add the following:

Added, and tagged on a DTSI fix for this SoC I noticed while looking at it for this PR.

**Review by anangl — APPROVED** [review]

**Pull request merged**